### ISSUE TITLE
System.Net.Http.Tests.WarningHeaderValueTest.GetWarningLength_DifferentInvalidScenarios_AllReturnZero failing in CI

### ISSUE THREAD

**Issue by jakobbotsch** [dotnet/runtime#85687]
Example run: https://dev.azure.com/dnceng-public/public/_build/results?buildId=259786&view=results
win-x64 System.Net.Http.WinHttpHandler.Unit.Tests
Log: https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-85323-merge-86f225c0d2434d34b0/System.Net.Http.Unit.Tests/3/console.80af5888.log?helixlogtype=result

```scala
  Starting:    System.Net.Http.Unit.Tests (parallel test collections = on, max threads = 4)
    System.Net.Http.Tests.WarningHeaderValueTest.GetWarningLength_DifferentInvalidScenarios_AllReturnZero [FAIL]
      Assert.Equal() Failure
      Expected: 0
      Actual:   11
      Stack Trace:
        /_/src/libraries/System.Net.Http/tests/UnitTests/Headers/WarningHeaderValueTest.cs(271,0): at System.Net.Http.Tests.WarningHeaderValueTest.CheckInvalidWarningViaLength(String input, Int32 startIndex)
        /_/src/libraries/System.Net.Http/tests/UnitTests/Headers/WarningHeaderValueTest.cs(175,0): at System.Net.Http.Tests.WarningHeaderValueTest.GetWarningLength_DifferentInvalidScenarios_AllReturnZero()
           at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
        /_/src/libraries/System.Private.CoreLib/src/System/Reflection/MethodInvoker.cs(59,0): at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
  Finished:    System.Net.Http.Unit.Tests
```

@tannergooding This looks like fallout from #85536, can you take a look?

The miscompilation happens inside `System.Net.Http.Headers.WarningHeaderValue:TryReadCode(System.String,byref,byref)`: https://www.diffchecker.com/dZckckLw/

Repros without stress too.

**Comment by jakobbotsch** [comment]
I wonder if this could be the cause for #85659 and #85637 as well, those are on xarch too.

**Comment by jakobbotsch** [comment]
Fixed by #85714
